x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_style_property.js

```diff
@@ -30,8 +35,61 @@ export class DynamicStyleProperty extends AbstractStyleProperty {
     return fieldSource && field ? fieldSource.getValueSuggestions(field, query) : [];
   };
 
-  getFieldMeta() {
-    return this._getFieldMeta && this._field ? this._getFieldMeta(this._field.getName()) : null;
+  _getStyleMetaDataRequestId(fieldName) {
```

Comment:
Since you have broken this into a seperate method, you can avoid all the nesting

```
_getStyleMetaDataRequestId(fieldName) {
    if (this.getFieldOrigin() === FIELD_ORIGIN.SOURCE) {
      return SOURCE_META_ID_ORIGIN;
    }
      
   const join = this._layer.getValidJoins().find(join => {
      return join.getRightJoinSource().hasMatchingMetricField(fieldName);
    });
    return join ? join.getSourceMetaDataRequestId() : null;
  }
```